os/arch/arm/src/s5j/s5j_i2s.c

```diff
@@ -1600,47 +1600,47 @@ static int i2s_pause(struct i2s_dev_s *dev)
 {
 	struct s5j_i2s_s *priv = (struct s5j_i2s_s *)dev;
 	irqstate_t flags;
+	volatile u32 reg = 0;
 	DEBUGASSERT(priv);
 
+	reg = getreg32(priv->base + S5J_I2S_CON);
+	flags = irqsave();
 #ifdef I2S_HAVE_TX_P
-	if (priv->txpenab) {
-		flags = irqsave();
+	if (priv->txpenab && reg & I2S_CR_TXDMACTIVE) {
 		modifyreg32(priv->base + S5J_I2S_CON, 0, I2S_CR_TXCHPAUSE);
```

Comment:
separate conditions with brackets. Logical and conditional expression evaluation will be easy to understand

Reply:
ok done